packages/pricing/src/models/price-list.ts

```diff
@@ -0,0 +1,79 @@
+import { generateEntityId } from "@medusajs/utils"
+import {
+  BeforeCreate,
+  Cascade,
+  Entity,
+  Enum,
+  Index,
+  OneToMany,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import MoneyAmount from "./money-amount"
+import { PriceListStatus, PriceListType } from "@medusajs/types"
+
+type OptionalRelations = "prices"
+type OptionalFields =
+  | "type"
+  | "status"
+  | "starts_at"
+  | "ends_at"
+  | "created_at"
+  | "updated_at"
+
+@Entity()
+class PriceList {
+  [OptionalProps]?: OptionalRelations | OptionalFields
+
+  @PrimaryKey({ columnType: "text" })
+  id!: string
+
+  @Property({ columnType: "text" })
+  name: string
+
+  @Property({ columnType: "text" })
+  description: string
+
+  @Enum({
+    items: () => PriceListType,
+    default: PriceListType.SALE,
+  })
+  type: PriceListType
+
+  @Enum({
+    items: () => PriceListStatus,
+    default: PriceListStatus.DRAFT,
+  })
+  status: PriceListStatus
+
+  @Property({ columnType: "timestamptz", nullable: true })
+  starts_at: Date | null
+
+  @Property({ columnType: "timestamptz", nullable: true })
+  ends_at: Date | null
+
+  @OneToMany({ entity: () => MoneyAmount, mappedBy: "price_list", orphanRemoval: true })
+  prices: MoneyAmount[]
```

Comment:
q: now that we've derailed from the core, should we just call these money_amounts?

Reply:
we could. I think calling them "prices" for at price-list is still reasonable but I'll happily name them "money_amounts" if we wish 😄

Reply:
Now that I'm writing the price selection, it might be better to push this through as money amounts, since we have price set, calculated price, price rules and the lot, calling it just price will be hella confusing.

I'm happy to assist in this if you'd like

Reply:
+1

Reply:
On second thought, since PriceLists are user-facing, wouldn't it make sense to use `prices`? 

Similar to how we are querying money amounts using a `calculate**Prices**`-method.